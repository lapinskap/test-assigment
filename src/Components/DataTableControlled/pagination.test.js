import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Pagination from './pagination';
import '../../tests/setupTests';

describe('<DataTableControlled />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(<Pagination
        pages={1}
        pageSize={10}
        availablePageSizes={[10, 20]}
        onPageSizeChange={() => {}}
        onPageChange={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly fires methods', () => {
    const onPageChange = jest.fn();
    const onPageSizeChange = jest.fn();

    const wrapper = shallow(<Pagination
      pages={3}
      pageSize={10}
      availablePageSizes={[10, 20]}
      onPageSizeChange={onPageSizeChange}
      onPageChange={onPageChange}
    />);

    const previousButton = wrapper.find('button').at(0);
    const nextButton = wrapper.find('button').at(1);
    const input = wrapper.find('input');
    const select = wrapper.find('select');

    previousButton.simulate('click');
    expect(onPageChange).toHaveBeenCalledTimes(1);
    nextButton.simulate('click');
    expect(onPageChange).toHaveBeenCalledTimes(2);
    input.prop('onChange')({ target: { value: 2 } });
    expect(onPageChange).toHaveBeenCalledTimes(3);
    select.prop('onChange')({ target: { value: 2 } });
    expect(onPageSizeChange).toHaveBeenCalledTimes(1);
  });
});
