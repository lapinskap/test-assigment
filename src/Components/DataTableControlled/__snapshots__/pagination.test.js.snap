// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTableControlled /> renders correctly with props 1`] = `
<div
  className="ReactTable"
>
  <div
    className="pagination"
  >
    <div
      className="pagination-bottom w-100"
    >
      <div
        className="-pagination"
      >
        <div
          className="-previous"
        >
          <button
            className="-btn"
            data-t1="gridPreviousPage"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Poprzednia
          </button>
        </div>
        <div
          className="-center"
        >
          <span
            className="-pageInfo"
          >
            Strona
            <div
              className="-pageJump"
            >
              <input
                aria-label="jump to page"
                data-t1="gridPage"
                onChange={[Function]}
                type="text"
                value={1}
              />
            </div>
            z
            <span
              className="-totalPages"
            >
              1
            </span>
          </span>
          <span
            className="select-wrap -pageSizeOptions"
          >
            <select
              aria-label="rows per page"
              data-t1="gridPageSize"
              onChange={[Function]}
              value={10}
            >
              <option
                value={10}
              >
                10
                 
                wpisów
              </option>
              <option
                value={20}
              >
                20
                 
                wpisów
              </option>
            </select>
          </span>
        </div>
        <div
          className="-next"
        >
          <button
            className="-btn"
            data-t1="gridNextPage"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Następna
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
